src/odemis/acq/feature.py

```diff
@@ -119,3 +145,412 @@ def load_project_data(path: str) -> dict:
             f.streams.value.extend(data_to_static_streams(open_acquisition(fname)))
 
     return {"overviews": overview_data, "features": features}
+
+
+def add_feature_info_filename(feature: CryoFeature, filename: str) -> str:
+    """
+    Add details of the given feature and the counter at the end of the given filename.
+    :param feature: the feature to add to the filename
+    :param filename: filename given by user
+    """
+    path_base, ext = splitext(filename)
+    feature_name = feature.name.value
+    feature_status = feature.status.value
+
+    path, basename = os.path.split(path_base)
+    ptn = f"{basename}-{feature_name}-{feature_status}-{{cnt}}"
+
+    return create_filename(path, ptn, ext, count="001")
+
+# To handle the timeout error when the stage is not able to move to the desired position
+# It logs the message and raises the MoveError exception
+class MoveError(Exception):
+    pass
+
+# TODO: generalise this function, and move it to the move module.
+# Currently, the _getDistance method is not available here.
+def estimate_stage_movement_time(start_pos: Tuple[float], end_pos: Tuple[float],
+                                 move_speed: float = MOVE_SPEED_DEFAULT) -> float:
+    """
+    Estimation the time taken by the stage to move from start to end position.
+    :param start_pos: the start position of the stage (x, y, z)
+    :param end_pos: the end position of the stage (x, y, z)
+    :param move_speed: the speed of the stage in m/s
+    :return: time to move the stage between two points in s.
+    """
+    diff = [abs(target - current) for target, current in zip(end_pos, start_pos)]
```

Comment:
I am kind of surprised we don't have some geometry utility in the codebase. Point to point distance seems pretty reusable to me.

Reply:
I agree, we should have a more general util for this. @pieleric I'm not sure if you know if we already do?

There is a _getDistance method in the posture manager class, but that relies on having the posture manager. We do have that here, so I can update this, but it would be good to just have a standalone function
https://github.com/patrickcleeve2/odemis/blob/f644906a104841474a172bc49e5552d9cd20cba7/src/odemis/acq/move.py#L157-L158